src/hypothesis/internal/compat.py

```diff
@@ -502,10 +502,16 @@ def implements_iterator(it):
 
 if PY2:
     def floor(x):
-        return int(math.floor(x))
+        # `int(math.floor(x))` fails for negative long integers
```

Comment:
What sort of failure mode are you seeing?

I see drift from the true value, but I think that’s an inevitability of computer-based arithmetic, and eventually an OverflowError – but the latter is true of Python 3 as well.